upload/serializers.py

```diff
@@ -133,13 +133,12 @@ class Meta:
         )
 
     def create(self, validated_data):
-        commit = Commit.objects.filter(
-            repository=validated_data.get("repository"),
-            commitid=validated_data.get("commitid"),
-        ).first()
-        if commit:
-            return commit
-        return super().create(validated_data)
+        repo = validated_data.pop("repository", None)
```

Comment:
Are you sure this is a `Repository` and not `RepositorySerializer`?

Reply:
It's because we pass `repository` to the `serializer.save` call [here](https://github.com/codecov/codecov-api/blob/b99b186333870d5ac8add76c7abe28708f3b87c0/upload/views/commits.py#L66) which will then get put into the dict that is passed to this `create` [here](https://github.com/encode/django-rest-framework/blob/master/rest_framework/serializers.py#L746-L775)

Reply:
perfection, thank you for checking that